src/Server.cpp

```diff
@@ -69,6 +167,7 @@ void Server::startPollEventLoop()
 										 _connections[i].getResponse());
 						if (_connections[i].getHasFinishedReading() && _connections[i].getHasDataToSend())
 							_FDs[i].events = POLLOUT;
+						CGIMonitor(_connections[i]);
```

Comment:
For me the monitor, i.e. the waitpid needs to be set, when we are done with poll, i.e. the very next line after the if/else conditions checking `ret` the returned value of poll. Probably I said at the end of 'hanldeConnection' in my message, but actually I meant 'just after poll' or right before going back to poll (after poll).

Reply:
I'm reviewing this again, while closing it, trying to understand if it could make sense to waitpid for everyone in the same else branch where we call handleConnection. Without re-checking my solution again I would say that if we are waiting for a CGI to complete we don't want to go necessarly thorough handleConnection is no CGI is returned and to check if it returned we want to go through waitpid, so they are two different things. When CGI is returned we want go back into handleConnection. But maybe you can also thing about it.